src/core/pointcloud/qgspointcloudrenderer.cpp

```diff
@@ -141,6 +141,16 @@ void QgsPointCloudRenderer::setMaximumScreenErrorUnit( QgsUnitTypes::RenderUnit
   mMaximumScreenErrorUnit = unit;
 }
 
+int QgsPointCloudRenderer::pointBudget() const
+{
+  return mPointBudget;
+}
+
+void QgsPointCloudRenderer::setPointBudget( int budget )
+{
+  mPointBudget = budget;
+}
```

Comment:
As said above, I would suggest to remove point budget from 2D renderer and only keep it for 3D renderer